### DISCUSSION TITLE
psutil.AccessDenied when python manager.py start

### DISCUSSION THREAD

**Discussion by joyarzun** [swar/Swar-Chia-Plot-Manager#302] [Q&A]
Hi

I've read a lot but I don't pick the right solution. I followed the instructions 100 times but the issue is still happening:

```
(venv) PS C:\Users\jonsxaero\Downloads\Swar-Chia-Plot-Manager-main\Swar-Chia-Plot-Manager-main> python manager.py start
Traceback (most recent call last):
  File "C:\Users\jonsxaero\Downloads\Swar-Chia-Plot-Manager-main\Swar-Chia-Plot-Manager-main\venv\lib\site-packages\psutil\_pswindows.py", line 780, in cmdline
    ret = cext.proc_cmdline(self.pid, use_peb=True)
PermissionError: [WinError 5] Access is denied: '(originated from OpenProcess)'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\jonsxaero\Downloads\Swar-Chia-Plot-Manager-main\Swar-Chia-Plot-Manager-main\venv\lib\site-packages\psutil\_pswindows.py", line 679, in wrapper
    return fun(self, *args, **kwargs)
  File "C:\Users\jonsxaero\Downloads\Swar-Chia-Plot-Manager-main\Swar-Chia-Plot-Manager-main\venv\lib\site-packages\psutil\_pswindows.py", line 695, in wrapper
    return fun(self, *args, **kwargs)
  File "C:\Users\jonsxaero\Downloads\Swar-Chia-Plot-Manager-main\Swar-Chia-Plot-Manager-main\venv\lib\site-packages\psutil\_pswindows.py", line 783, in cmdline
    ret = cext.proc_cmdline(self.pid, use_peb=False)
PermissionError: [Errno 13] assume access denied (originated from NtQueryInformationProcess(ProcessBasicInformation) -> STATUS_NOT_FOUND)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\jonsxaero\Downloads\Swar-Chia-Plot-Manager-main\Swar-Chia-Plot-Manager-main\manager.py", line 29, in <module>
    start_manager()
  File "C:\Users\jonsxaero\Downloads\Swar-Chia-Plot-Manager-main\Swar-Chia-Plot-Manager-main\plotmanager\library\utilities\commands.py", line 20, in start_manager
    if get_manager_processes():
  File "C:\Users\jonsxaero\Downloads\Swar-Chia-Plot-Manager-main\Swar-Chia-Plot-Manager-main\plotmanager\library\utilities\processes.py", line 32, in get_manager_processes
    if not _contains_in_list('python', process.cmdline(), case_insensitive=True) or \
  File "C:\Users\jonsxaero\Downloads\Swar-Chia-Plot-Manager-main\Swar-Chia-Plot-Manager-main\venv\lib\site-packages\psutil\__init__.py", line 677, in cmdline
    return self._proc.cmdline()
  File "C:\Users\jonsxaero\Downloads\Swar-Chia-Plot-Manager-main\Swar-Chia-Plot-Manager-main\venv\lib\site-packages\psutil\_pswindows.py", line 681, in wrapper
    raise convert_oserror(err, pid=self.pid, name=self._name)
psutil.AccessDenied: psutil.AccessDenied (pid=1396, name='python3.8')
```

At windows 10 pro with chia_location: C:\Users\jonsxaero\AppData\Local\chia-blockchain\app-1.1.4\resources\app.asar.unpacked\daemon\chia.exe

I installed python 3.9 from the official website

My steps used in powershell(administrator) are:

```
cd  C:\Users\jonsxaero\Downloads\Swar-Chia-Plot-Manager-main\Swar-Chia-Plot-Manager-main
python -m venv venv
venv\Scripts\activate
pip install -r requirements.txt
python manager.py start
```

The pip install finished with:

`Successfully installed PyYAML-5.4.1 certifi-2020.12.5 chardet-4.0.0 dateparser-1.0.0 discord-notify-1.0.0 idna-2.10 playsound-1.2.2 psutil-5.8.0 python-dateutil-2.8.1 python-pushover-0.4 pytz-2021.1 regex-2021.4.4 requests-2.25.1 six-1.16.0 tzlocal-2.1 urllib3-1.26.4`

Something interesting:

![image](https://user-images.githubusercontent.com/828211/117913017-9348af80-b2ae-11eb-95b4-e12853a9463a.png)

Maybe the psutil has an issue with windows

**Comment by Allram** [comment]
Try to run it with this command:
`python .\manager.py start`

**Comment by Taipan24** [comment]
I got similar messages after doing a Windows update today,
and now I can run it only with administrator privileges.

**Comment by joyarzun** [comment]
> Try to run it with this command:
> `python .\manager.py start`

![image](https://user-images.githubusercontent.com/828211/117987257-4db8d080-b308-11eb-889d-0a6de08d80ca.png)

Same errors. Another suggestion? My background is more related to Linux so I'm lost at this point

**Comment by joyarzun** [comment]
> I got similar messages after doing a Windows update today,
> and now I can run it only with administrator privileges.

I have a recent windows installation and furthermore, I updated it recently so can be windows?

I ran it with administrator Powershell

**Comment by chenseanxy** [comment]
Hey, can you re-run it and check which process exactly is causing psutil.AccessDenied (by using pid)? Mine is usually start_full_node.

**Comment by swar** [comment]
Try reinstalling another version of Python. That seems to solve the issue for some people.

**Comment by Dark-Sider** [comment]
Hi,

I ran into the same problem. I traced the exception. It appears that the psutils is unable to read the names of certain processes and thus throws a psutils.AccessDenied exception. This exception is not caught by the current code, so the manager crashes. In my cases the process in question was Microsofts XBOX Live gaming bar.

I fixed the issue for myself by editing the processes.py file: (see my commit here https://github.com/swar/Swar-Chia-Plot-Manager/pull/315/commits/2443f60ae7e430fe5e9db6148fb151f53eaa3de0).

I also issued a pull request, which was denied: https://github.com/swar/Swar-Chia-Plot-Manager/pull/315

I don't understand the reasoning behind the author's decision not to include this fix. My fix does not skip any processes that could have been otherwise read, it just prevents the script from crashing. When it comes down to choice between a crash and exception handling it should always be on the side of exception handling :)

Also installing different versions of Python will not solve an access denied error by windows ACL.

regards.
Dark-Sider

**Reply by chenseanxy** [reply]
Hi there, PR #147 is still open (for your reference)

**Reply by Dark-Sider** [reply]
Thx, I didn't even notice that  https://github.com/swar/Swar-Chia-Plot-Manager/pull/147 existed - my bad. The proposed fix seems identical :) 

I'm running with this fix without any issues - and I still don't see how this fix can cause any issues.
